### ISSUE TITLE
Creating MO image crashes 86Box

### ISSUE THREAD

**Issue by mph2016** [86Box/86Box#930]
**Describe the bug**
Attempting to create a new MO image using the UI crashes 86Box

**To Reproduce**
1. Go to menu -> MO or right-click on the icon
2. Click on 'create new image'
3. use the UI to select a folder, write a file-name and press ok
4. The UI begins the process of writing out an image file and stops about 1/3rd of the way and then 86Box "goes away" (crashes)

**Expected behavior**
I expected the file to complete writing and then attatch to the emulated machine.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - 86Box version: v2.10 build 2474
 - Build type: Regular build

**Additional context**
The file seems to write out about 98 megabytes before crashing.

**Comment by ppgrainbow** [comment]
Can you post your 86Box.cfg file to find out what configuration did you use to create a magnetic optical disk image and what guest operating system did you use to create it?

**Comment by mph2016** [comment]
It doesn't seem to have anything to do with the guest operating system. I tried this while booted inside of FreeBSD 2.0 but also while in the bios (so that I could avoid having to sit through a long fsck in the guest).  

I'm doing this in the host UI, not in the guest.

Here is the cfg file and nvr/ram I'm using. 
[FreeBSD 2.0.zip](https://github.com/86Box/86Box/files/4935097/FreeBSD.2.0.zip)

**Comment by ppgrainbow** [comment]
Thank you for the heads up. Let's hope that the magneto-optical drive creation gets fixed to a point where it is usable soon. :)

**Comment by OBattler** [comment]
Try now, see if it works correctly now with all the changes.

**Comment by ppgrainbow** [comment]
The crash has been fixed. However, for some reason, I can't mount the magneto-optical disk image and every time that I do so, the size of the MO image always defaults to 100 MB.

![20200719-042802-960](https://user-images.githubusercontent.com/32658796/87867072-ed524000-c93d-11ea-96c5-d2f5be8f2e27.png)

The config file can be found here: https://pastebin.com/7SSrNdF7

The MO device emulated is a IBM 0632-AAA (650 MB optical) and is located on SCSI ID #1.

**Comment by mph2016** [comment]
I'm having the same result. 86Box doesn't crash, but the dialog box "goes away" and leaves an almost empty (like 4k or so) file but doesn't mount it.  I had mine set to SCSI too, and tried different models and sizes (but got the same results).

I tried it on build 2482, regular (non-optimized) build; windows 10.

**Comment by ppgrainbow** [comment]
I have also tried the ATAPI versions and still the same result. :\

**Comment by TC1995** [comment]
I think it's due to this:
pbar_max = dp->sectors >> 11;
which creates a tiny file compared to what the user expects.

**Comment by ppgrainbow** [comment]
Uh oh! That doesn't sound too good. :(

I sure hope that there is a way to fix the problem.

**Comment by ppgrainbow** [comment]
Okay, the MO disk creation works now as it should.

However, I still can't mount the MO image upon inserting it. I can mount floppiess, CD-ROMs and ZIP drives though.

Is there a way how this can be fixed? :(

**Comment by mph2016** [comment]
The dialog doesn't quite complete, but it seems to make the proper sized files.
![19july2020](https://user-images.githubusercontent.com/17302103/87883315-fb8e7380-c9b2-11ea-8aec-2dc9fd3078e9.PNG)
(that's the results of my trying to create iso 10090 and iso 13963 images)

The "create image" process doesn't mount them in 86box, though. I don't seem to be able to load them in the emulator after they're created. 

Should I close this ticket (since 86box no longer crashes) and open a new one for that since it's a different issue?

**Comment by OBattler** [comment]
Eh, I see no problem if discussion about it continued here.

**Comment by mph2016** [comment]
I've tested it some more and was able to make (and mount) a couple of different images in a couple of different models. 

Trying to make a 128 or a 230 image the dialog doesn't seem to finish, though it seems to write out an image file. I can't mount the image files.  

I am able to successfully create a 540, 640 and a 1.3gb (gigamo) image and mount it in 86Box. 

When I try to create a 2.3 (gigamo 2) image it crashes 86Box.

I was able to create a 600mb and a 640mb 5.25' image, but when I tried to make a 1gb 5'25 image the dialog vanished and there was an incomplete (400meg or so) file. I had the same result with the 1.3gb 5.25 image except it crashed after writing approximately 600 megs. 

I have "86box Magneto Optical 1.00" selected for the model in settings.

**Comment by OBattler** [comment]
I seem to have none of these problems (I just tried), so maybe try with the latest 86Box and see if they still happen for you.

**Comment by OBattler** [comment]
Well, I'm closing this, it can be reopened if problems resurface.

**Comment by mph2016** [comment]
I tried it with the same model ("86box Magneto Optical 1.00")

The 128 dialog box went away before it finished; I'm not sure if it wrote out a complete a complete file or not; pretty much the same for the 256mb tape. 
![MO_generated_Files-26Oct2020](https://user-images.githubusercontent.com/17302103/97213443-7c10fd00-1776-11eb-88ac-c4b6289451ef.PNG)

I haven't had time to test the rest yet -I just saw the notifications for this pr

**Comment by mph2016** [comment]
I was able to mount the images; but if they're not complete I'm not sure what that means. :|

**Comment by mph2016** [comment]
I'm still getting the same results today that I did before.  
Here's the images I was able to generate on my computer, named after the sizes I picked:
![MagenticOpticalDrive_Test-26Oct2020](https://user-images.githubusercontent.com/17302103/97214825-7d432980-1778-11eb-9477-2c9173986190.PNG)

They're still not being written out -at least on my computer. 

This is with build 2617, unoptimized, windows 10 host.
